### DISCUSSION TITLE
How to modify object properties ?

### DISCUSSION THREAD

**Discussion by johnbhlm** [StanfordVL/BEHAVIOR-1K#1003] [Q&A] [answered]
Hi,
thanks for great work!

We want to modify the properties of the object, for example, say after loading the .USD file, add rigid body properties and collision properties on the object, or change the mesh's “Visual” property “guide” to “default”. so how should I add or modify the code to realize this function?

Looking forward to your reply.

**Comment by hang-yin** [comment, marked as answer]
Hi @johnbhlm , if I'm understanding your question correctly, you can first grab base link prim of the object:
```
import omnigibson.lazy as lazy
obj_prim = lazy.omni.isaac.core.utils.prims.get_prim_at_path(prim_path="/World/scene_0/object_name/base_link")
```

Then, you can directly set attributes on this prim, for example:
```
obj_prim.GetAttributes() # this gives you all available attributes on this prim

obj_prim.GetAttribute('purpose').Get() # this gives you the value of a specific attribute

obj_prim.GetAttribute('purpose').Set("default") # this sets the attribute
```

Let me know if this helps!
